### PR TITLE
Add credentials variables to env variables for metrics exporter

### PR DESCRIPTION
**Pull request by tielou**
Hi

**Description of the change**

This adds environment variables for the metrics exporter as described over here: https://github.com/bitnami/charts/pull/7311

**Benefits**

One can use a different user than root for the metrics exporter.

**Possible drawbacks**

none

**Applicable issues**

https://github.com/bitnami/charts/issues/7310

**Additional information**

none

### CODE REVIEW

**Comment by yilmi** [comment]
Hi @tielou,

Thanks a mil for this PR!

Could you please add the users creation step to your PR?

This way the we can have the metrics user created at runtime, and it will cover the changes from [your chart changes](https://github.com/bitnami/charts/issues/7310#issuecomment-908461737).

Sharing some references to help:
- Where are users created in our scripts: https://github.com/bitnami/bitnami-docker-mongodb/blob/master/4.4/debian-10/rootfs/opt/bitnami/scripts/libmongodb.sh#L501
- Minimum permissions required for the metrics user: https://stuff.21zoo.com/posts/prometheus-mongodb-exporter-user-permissions/

Thanks,
Yass

**Comment by yilmi** [comment]
Hi @tielou,

Let us know if you have any questions for this change and we'll do my best to help you get your contribution through!

Thanks,
Yass

**Comment by tielou** [comment]
Hi @yilmi,

sorry I was out sick last week, therefore the delayed reply. 
I added the creation to the PR.
But I also thought about cases, where users don't want to create the metrics user but just specify it - let's say someone switches to the MongoDB chart and restores a dump which already includes a metrics user. Since those scripts are just executed at initialisation this shouldn't make a difference right?

**Comment by yilmi** [comment]
Hi @tielou, I hope you're feeling better!

You are correct, `mongodb_create_users()` is only called if no data is found as defined [here](https://github.com/bitnami/bitnami-docker-mongodb/blob/master/4.4/debian-10/rootfs/opt/bitnami/scripts/libmongodb.sh#L1094). I'll review your changes and hopefully we can merge both your changes early next week.

File: 4.4/debian-10/rootfs/opt/bitnami/scripts/libmongodb.sh [hunk 1]
```diff
@@ -517,6 +517,16 @@ EOF
         result=$(
             mongodb_execute 'root' "$MONGODB_ROOT_PASSWORD" "" "127.0.0.1" <<EOF
 db.getSiblingDB('$MONGODB_DATABASE').createUser({ user: '$MONGODB_USERNAME', pwd: '$MONGODB_PASSWORD', roles: [{role: 'readWrite', db: '$MONGODB_DATABASE'}] })
+EOF
+        )
+    fi
+
+    if [[ -n "$MONGODB_METRICS_USERNAME" ]] && [[ -n "$MONGODB_METRICS_PASSWORD" ]]; then
```
Comment (yilmi) [inline 4.4/debian-10/rootfs/opt/bitnami/scripts/libmongodb.sh:524]: There is a parameter called ALLOW_EMPTY_PASSWORD, which has to be set to yes, to allow the creation of users without any password defined.

This is not a problem for the Charts use case, but if someone consumes the container directly (docker-compose or other), the ALLOW_EMPTY_PASSWORD setting won't be honored for this used.

Would you mind adding something like this after line 91 to cover this user?
```bash
    if ! is_boolean_yes "$ALLOW_EMPTY_PASSWORD" && [[ -n "$MONGODB_METRICS_USERNAME" ]] && [[ -z "$MONGODB_METRICS_PASSWORD" ]]; then
        error_message="The MONGODB_METRICS_PASSWORD environment variable is empty or not set. Set the environment variable ALLOW_EMPTY_PASSWORD=yes to allow the container to be started with blank passwords. This is only recommended for development."
        print_validation_error "$error_message"
    fi
```

**Review by yilmi — CHANGES_REQUESTED** [review]

**Comment by yilmi** [comment]
Looks good to me, just added a comment regarding ALLOW_EMPTY_PASSWORD for cases others than helm charts. We should be able to merge after that. Thanks!

**Review by yilmi — APPROVED** [review]

**Comment by yilmi** [comment]
Your changes should be released tonight, if you could test your PR from https://github.com/bitnami/charts/pull/7311 and let us know if it works as expected that would be amazing 👍

**Pull request merged**